### ISSUE TITLE
Alternate ways to specify JVM and Scala versions 

### ISSUE THREAD

**Issue by aschofie** [VirtusLab/scala-cli#2243]
Unless I am satisfied with whatever is the default JVM and the latest scala version at the moment, I have to specify the JVM version and Scala version as command-line options, or in the case of the scala version I can also do:

//> using scala <scala-version>

It would be great if we could specify in a config file and as environment variables what the "default" versions should be.

**Comment by lwronski** [comment]
@aschofie Thanks for reporting,  It appears to be a similar idea to the feature request mentioned in this issue:  https://github.com/VirtusLab/scala-cli/issues/2190.

**Comment by aschofie** [comment]
@lwronski Yes this is very similar to the #2190 except that it also wants the functionality extended to being able to set a default jvm version.

**Comment by lwronski** [comment]
@aschofie, can you provide more details about your use case? I would like to understand why this feature would be useful for you. Are you referring to only the local config or to global config?

**Comment by aschofie** [comment]
@lwronski I had envisioned only the local config-- the user's environment variables and/or some file in the user's home directory.  Unless a script has:

//> using javaHome
//> using scala

We don't know what version it was developed, tested, and run under after some time has passed.  Is the problem I'm experiencing now because the java and scala versions are different, or something else?  Ideally the versions would always be specified, but those 2 lines start to make scala-cli scripts feel rather boilerplate-ish.

In general, it is very difficult to piece together when scala-cli will check for a new java or scala version.  The behavior AFAICT on what it will do if it has java and scala versions downloaded but cannot reach the internet is also undefined.  Does it make a network request on every scala-cli invocation?  Maybe the information is all in the docs, but it is scattered and has less clarity than I would like.  Maybe all my issues could be solved by having all of this information in a single page in the documentation titled: 

"What exact versions of java and scala will scala-cli use: all the ways to set them, when scala-cli will talk to the network, and what will happen if it can't."

Huge bonus if scala-cli also could be prevented from talking to the network under any circumstances.

**Comment by lwronski** [comment]
> What exact versions of java will scala-cli use:

To use a specific version of Java, it's simpler to set Java 17 or higher on your `PATH`. ScalaCLI guarantees that if Java >17 is available in the PATH, it's used to compile your project and run `bloop`. For instance, you can set up the Java path in Github CI, which can be downloaded from a user-specified URL

```
      - uses: coursier/setup-action@6a582d7f7292a865e72c497ca64c3ef447cdb6c7
        with:
          apps: ""
          jvm: "https://github.com/graalvm/graalvm-ce-builds/releases/download/vm-22.2.0/graalvm-ce-java17-darwin-aarch64-22.2.0.tar.gz"
```

> What exact versions of java will scala-cli use:

ScalaCLI guarantees the default Scala version remains constant within a specific Scala CLI version. However, the newer ScalaCLI might use a newer Scala version as a default. For reproducible builds, please specify the complete full version of Scala

```
//> using scala 3.3.0
```

> Does it make a network request on every scala-cli invocation?

This relates to a different issue at https://github.com/VirtusLab/scala-cli/issues/1421. The primary objective there is to ensure that `scala-cli` doesn't download something from the internet. Currently, ScalaCLI doesn't offer this guarantee, but we're looking to incorporate it.
